vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/ad_targeting_segment_util.cc

```diff
@@ -48,36 +48,48 @@ SegmentList GetParentSegments(const SegmentList& segments) {
   return parent_segments;
 }
 
-bool ShouldFilterSegment(const std::string& segment) {
-  // If passed in segment has a sub segment and the current filter does not,
-  // check if it's a child of the filter. Conversely, if the passed in segment
-  // has no sub segment but the current filter does, it can't be a match at all
-  // so move on to the next filter. Otherwise, perform an exact match to
-  // determine whether or not to filter the segment
+bool HasChildSegment(const std::string& segment) {
+  const std::vector<std::string> components = SplitSegment(segment);
+  DCHECK(!components.empty());
+
+  if (components.size() == 1) {
+    return false;
+  }
+
+  return true;
+}
 
-  const std::vector<std::string> segment_components = SplitSegment(segment);
+bool ParentSegmentsMatch(const std::string& lhs, const std::string& rhs) {
+  const std::vector<std::string> lhs_segment_components = SplitSegment(lhs);
+  const std::string lhs_parent_segment = lhs_segment_components.front();
```

Comment:
Maybe we should add DCHECK here that components are not empty

Reply:
Fixed